File: server/scripts/init-db.js
/**
 * Инициализация базы данных
 * 
 * Запуск: node init-db.js
 */

const Database = require('better-sqlite3');
const path = require('path');
const fs = require('fs');

// Путь к БД
const dbPath = path.join(__dirname, '..', 'tasksbd.db');
console.log('Создание базы данных:', dbPath);

// Удаляем старую БД если есть
if (fs.existsSync(dbPath)) {
  fs.unlinkSync(dbPath);
  console.log('✓ Старая БД удалена');
}

// Создаем новую БД
const db = new Database(dbPath);

// Читаем SQL скрипт
const sqlPath = path.join(__dirname, '..', 'database', 'init.sql');
const sqlScript = fs.readFileSync(sqlPath, 'utf-8');

// Выполняем SQL скрипт
db.exec(sqlScript);

console.log('✅ База данных создана успешно!');

// Проверяем таблицы
const tables = db.prepare("SELECT name FROM sqlite_master WHERE type='table'").all();
console.log('\nСозданные таблицы:');
tables.forEach(table => console.log('  -', table.name));

db.close();

File: server/scripts/init_db.py
# -*- coding: utf-8 -*-
"""
Инициализация базы данных
"""

import sqlite3
import sys
from pathlib import Path

# Путь к БД
script_dir = Path(__file__).parent
db_path = script_dir.parent / 'tasksbd.db'

print(f"Создание базы данных: {db_path}")

# Создаем БД
conn = sqlite3.connect(str(db_path))
cursor = conn.cursor()

# Читаем SQL скрипт
sql_path = script_dir.parent / 'database' / 'init.sql'
with open(sql_path, 'r', encoding='utf-8') as f:
    sql_script = f.read()

# Выполняем SQL скрипт
cursor.executescript(sql_script)
conn.commit()
conn.close()

print("✅ База данных создана успешно!")


